From 158d896f29ff74006d632610cddca1d770a08c77 Mon Sep 17 00:00:00 2001
From: xianbing Zhu <xianbing.zhu@linux.alibaba.com>
Date: Fri, 31 May 2024 14:25:37 +0800
Subject: [PATCH 104/400] mmc:sdhci-of-dwcmshc: th1520 sdhci add fix io voltage
 1v8

In sdhci ops set_power SDHCI_POWER_CONTROL set 0 will
cause CTRL2 reg SDHCI_CTRL_VDD_180 filed cleared,so
add SDHCI_QUIRK_SINGLE_POWER_WRITE to avoid this.

Signed-off-by: xianbing Zhu <xianbing.zhu@linux.alibaba.com>
---
 drivers/mmc/host/sdhci-of-dwcmshc.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/drivers/mmc/host/sdhci-of-dwcmshc.c b/drivers/mmc/host/sdhci-of-dwcmshc.c
index 6457e503b98c..99eb33539728 100644
--- a/drivers/mmc/host/sdhci-of-dwcmshc.c
+++ b/drivers/mmc/host/sdhci-of-dwcmshc.c
@@ -719,7 +719,8 @@ static const struct sdhci_pltfm_data sdhci_dwcmshc_rk35xx_pdata = {
 
 static const struct sdhci_pltfm_data sdhci_dwcmshc_th1520_pdata = {
 	.ops = &sdhci_dwcmshc_th1520_ops,
-	.quirks = SDHCI_QUIRK_CAP_CLOCK_BASE_BROKEN,
+	.quirks = SDHCI_QUIRK_CAP_CLOCK_BASE_BROKEN |
+			SDHCI_QUIRK_SINGLE_POWER_WRITE,
 	.quirks2 = SDHCI_QUIRK2_PRESET_VALUE_BROKEN,
 };
 
@@ -895,7 +896,8 @@ static int dwcmshc_probe(struct platform_device *pdev)
 
 		if ((device_property_read_bool(dev, "mmc-ddr-1_8v")) |
 		    (device_property_read_bool(dev, "mmc-hs200-1_8v")) |
-		    (device_property_read_bool(dev, "mmc-hs400-1_8v")))
+		    (device_property_read_bool(dev, "mmc-hs400-1_8v")) |
+			(device_property_read_bool(dev, "io_fixed_1v8")))
 			priv->flags |= FLAG_IO_FIXED_1V8;
 		else
 			priv->flags &= ~FLAG_IO_FIXED_1V8;
-- 
2.43.0

